src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/StorePal.cs

```diff
@@ -264,15 +264,19 @@ private static void LoadMachineStores()
                             if (uniqueRootCerts.Add(cert))
                             {
                                 rootStore.Add(cert);
+                                continue;
                             }
                         }
                         else
                         {
                             if (uniqueIntermediateCerts.Add(cert))
                             {
                                 s_machineIntermediateStore.Add(cert);
+                                continue;
                             }
                         }
+
+                        cert.Dispose();
```

Comment:
It took me a minute to realize what was actually being Disposed here.  I'd probably have written it as `if (unique) { Add(); } else { Dispose(); }`, but this is fine... mind adding a comment to the effect of

```C#
// There's a good chance that we'll encounter duplicates on systems that have both one-cert-per-file
// and one-big-file trusted certificate stores. Anything that wasn't unique will end up here.
```